src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Traits;
+
+use Exception;
+use Illuminate\Database\Eloquent\Relations\Relation;
+
+trait Relationships
+{
+    /**
+     * If the field name is not a relationship method e.g: article_id,
+     * we try to find if this field has a relation defined.
+     *
+     * @param string $fieldName
+     * @return array|bool
+     */
+    public function checkIfFieldNameBelongsToAnyRelation($fieldName)
+    {
+        $relations = $this->getAvailableRelationsInModel();
+
+        if (empty($relations)) {
+            return false;
+        }
+
+        if (in_array($fieldName, array_column($relations, 'name'))) {
+            return array_filter($relations, function ($arr) use ($fieldName) {
+                if (isset($arr['name'])) {
+                    return $arr['name'] == $fieldName;
+                }
+
+                return false;
+            })[0];
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the user defined methods in model that return any type of relation.
+     *
```

Comment:
## Relationship Issue 12

Details about how this works here - https://github.com/Laravel-Backpack/CRUD/pull/2311#discussion_r381992855

Let's please better explain in the comments that in order for this to work, the relationships in the model have to have their return type defined (and give a few examples). I'd say:

```php
/**
     * Get the user defined methods in model that return any type of relation.
     * Only returns methods that have their return type explicitly specified. For example:
     * public function article() : BelongsTo {}
     * public function tags() : HasMany {}
     */
```

Also let's make a note for our generators to have models with return types by default (this is the note 😝 ).

Reply:
Confirmed. Leaving open so we include the note above in the docs.

Reply:
Done - docs here https://github.com/Laravel-Backpack/Docs/commit/65d899069600473e3ce7df601beab67c110bcb2d